docs/doc.main

```diff
@@ -146,12 +146,20 @@
 
 
   "General" => [
-     "General/styleguide.md",
      "General/other.md",
      "General/faq.md",
-		 "General/SubObjectIterator.md"
   ],
 
   "References" => "references.md",
   "Index" => "manualindex.md",
+  "Developers Corner" => [
+		"DevelopersCorner/intro.md",
+		"General" => [
+			"DevelopersCorner/styleguide.md",
+			"DevelopersCorner/build_documentation.md"
```

Comment:
Good idea to move `docs/README.md` here, but I would suggest to have the filename and the title of the document be related; so perhaps change the .md file to have title `# Building the OSCAR documentation` ?